compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/window/Popup.kt

```diff
@@ -16,14 +16,93 @@
 
 package androidx.compose.ui.window
 
+import androidx.compose.runtime.Composable
 import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.unit.IntRect
 import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.IntSize
 import androidx.compose.ui.unit.LayoutDirection
-import androidx.compose.ui.unit.height
-import androidx.compose.ui.unit.width
+
+/**
+ * Properties used to customize the behavior of a [Popup].
+ *
+ * @property focusable Whether the popup is focusable. When true, the popup will receive IME
+ * events and key presses, such as when the back button is pressed.
+ * @property dismissOnBackPress Whether the popup can be dismissed by pressing the back button.
+ * If true, pressing the back button will call onDismissRequest. Note that [focusable] must be
+ * set to true in order to receive key events such as the back button - if the popup is not
+ * focusable then this property does nothing.
+ * @property dismissOnClickOutside Whether the popup can be dismissed by clicking outside the
+ * popup's bounds. If true, clicking outside the popup will call onDismissRequest.
+ * @property securePolicy Policy for setting [android.view.WindowManager.LayoutParams.FLAG_SECURE] on the popup's
+ * window.
+ * @property excludeFromSystemGesture A flag to check whether to set the systemGestureExclusionRects.
+ * The default is true.
+ * @property clippingEnabled Whether to allow the popup window to extend beyond the bounds of the
+ * screen. By default the window is clipped to the screen boundaries. Setting this to false will
+ * allow windows to be accurately positioned.
+ * The default value is true.
+ * @property usePlatformDefaultWidth Whether the width of the popup's content should be limited to
+ * the platform default, which is smaller than the screen width.
+ */
+@Immutable
+class PopupProperties @ExperimentalComposeUiApi constructor(
```

Comment:
There was a concern by some guys that common classes shouldn't contain "ignored" parameters. Because when users write their application, they usually don't know which parameter is ignored on which platform, and don't understand why parameter doesn't do anything.

I don't know how to properly fix this issue yet. Only have some ideas and need to think about them more:

1. This is not actually an issue, leave it as it is in the current code
2. Make PopupProperties expect/actual. Keep in common properties that we can handle on all platforms.
3. Think about what the other non-Kotlin frameworks do? (they don't have expect/actual)
4. Keep in mind, that platforms can not only be separated by sourceSets (we can adapt Compose for JavaFX, Swing, IDEA, LWJGL platforms in the same JVM sourceset). Which properties can we always support in this case?

Reply:
2. It will make it not binary compatible constructor with Android. It will lead delay to 1.6 and prevent commonization of some material components until that

Reply:
@igordmn We would still want to provide the functionality on the platforms that do support it, right? Then we either go down the rabbit hole of providing multiple functions (which can become a complex mess) or document which functionality is available on which platforms.

Also, trying to only provide the lowest-common-denominator we can run into a problem when we release an API, and then add a new platform where some functionality is unavailable. We won't be able to remove it, since it's already been released...